src/pkg/cli/subscribe.go

```diff
@@ -66,20 +56,19 @@ func Subscribe(ctx context.Context, client client.Client, services []string) (<-
 			}
 
 			subStatus := SubscribeServiceStatus{
-				Name:   msg.GetName(),
-				Status: msg.GetStatus(),
-				State:  msg.GetState(),
+				Name:   msg.Name,
+				Status: msg.Status,
+				State:  msg.State,
 			}
 
-			servInfo := msg.GetService()
+			servInfo := msg.Service
 			if subStatus.Name == "" && (servInfo != nil && servInfo.Service != nil) {
 				subStatus.Name = servInfo.Service.Name
 				subStatus.Status = servInfo.Status
 				subStatus.State = convertServiceState(servInfo.Status)
 			}
 
 			statusChan <- subStatus
```

Comment:
The channel and go routine is not necessary, there is no concurrency need or synchronisation point, it is purely acting as a modified function call, we can either use the struct solution in (#518 - subscribe.go)[https://github.com/DefangLabs/defang/pull/518/files#diff-4736e1cbd4382619065b89be426ee362b9666ef6ef362021f0b7765f5e75da22], or come to think about it, we can just create a conversion function for serviceInfo to servicestatus and keep the rest in the commands.go loop.

Reply:
Agreed

Reply:
done in https://github.com/DefangLabs/defang/pull/556